scipy/stats/_survival.py

```diff
@@ -196,3 +189,60 @@ def _ecdf_right_censored(sample):
     sf = np.cumprod((n - d) / n)
     cdf = 1 - sf
     return ECDFResult(t, cdf, sf)
+
+
+def _kaplan_meier(sample):
+    tod = sample._uncensored  # time of "death"
+    tol = sample._right  # time of "loss"
+    times = np.concatenate((tod, tol))
+    died = np.asarray([1]*tod.size + [0]*tol.size)
+
+    # sort by times
+    i = np.argsort(times)
+    times = times[i]
+    died = died[i]
+    at_risk = np.arange(times.size, 0, -1)
+    return times, died, at_risk
+
+
+def _log_rank(sample):
+    times, died, at_risk = _kaplan_meier(sample)
+
+    idx_died = died.astype(bool)
+    times = times[idx_died]
+    at_risk = at_risk[idx_died]
+    return times, died, at_risk
+
+
+def log_rank(t_sample, t_control, e_sample, e_control):
```

Comment:
What about just accepting instances of `CensoredData`?

Reply:
I would prefer that too if you would be ok with it!

Reply:
I mean logically it could also be two ecdf results since this is about comparing 2 SF

Reply:
I thought about that originally. Yeah, it could accept either, but I don't think it should be required. Note that to get the `ECDF` results, they must have the original `CensoredData`s, so I'm not sure it buys us much.